packages/core/src/model.ts

```diff
@@ -6,24 +6,82 @@ import type {
   ExtractEvent,
   EventObject
 } from './types';
+import { mapValues } from './utils';
 
-export interface ContextModel<TContext, TEvent extends EventObject> {
+export interface ContextModel<
+  TContext,
+  TEvent extends EventObject,
+  TEM extends EventCreatorMap<TEvent> = EventCreatorMap<TEvent>
```

Comment:
I'm unsure if this should have a default here - if one currently has smth like this used anywhere:
```ts
type A = ContextModel<{ count: number }, { type: 'inc' } | { type: 'dec' }>
```
then they would suddenly get a guarantee that there is an `events` property available on `A` - whereas this is not true. It's only defined if the `ContextModel` comes from the `createModel` call with `eventCreators` provided.

To make this more correct I would try to restructure this to accept already computed `FullEventCreatorMap` and default this to `never`.  This way lone `ContextModel` usages wouldn't get this property but `createModel` could enhance the model with it.

Reply:
Maybe, but the main use-case is a model that is _not_ specified with generics (inferred from arguments instead).

Reply:
Yes, I understand that but up until now providing generics was the preferred way of using the model and thus this is a conceptually perfect code:
https://codesandbox.io/s/objective-ellis-oy5jq?file=/src/index.ts:526-601

With changes from this PR, the new property `.events` would slip into the model there and that would be incorrect as it wouldn't actually be available at runtime

Reply:
This is not a concern, because the inferred type becomes `never`:

<img width="541" alt="CleanShot 2021-03-13 at 20 44 33@2x" src="https://user-images.githubusercontent.com/1093738/111054765-1aa7ad00-843d-11eb-8b97-4eed7f284351.png">
<img width="576" alt="CleanShot 2021-03-13 at 20 45 41@2x" src="https://user-images.githubusercontent.com/1093738/111054766-1c717080-843d-11eb-93cb-6dc42921f264.png">

Reply:
Yep - the inferred type is correct. The demo showcases how the explicit usage of the Model type can go wrong though.

Reply:
Let's tackle that in a follow-up PR.

Reply:
The change is really simple ( https://github.com/davidkpiano/xstate/pull/1995 ) if we'd like to do it. Note also that creating a default `EventCreatorMap` has also produced unsafe code even for models with appropriate creators - since all arguments were typed as `any` in such a case.